File: noop-java/src/main/java/com/exyui/android/debugbottle/ui/BlockCanaryContext.java
package com.exyui.android.debugbottle.ui;

import android.content.Context;

/**
 * Created by yuriel on 9/7/16.
 */

public class BlockCanaryContext {
    public BlockCanaryContext(Object a) {}

    public int getConfigDuration() {
        return 99999;
    }

    public long getConfigBlockThreshold() {
        return 0;
    }

    public boolean isNeedDisplay() {
        return false;
    }

    public String getQualifier() {
        return "";
    }

    public String getUid() {
        return "";
    }

    public String getNetworkType() {
        return "";
    }

    public Context getContexr() {
        return null;
    }

    public String getLogPath() {
        return "";
    }

    public boolean zipLogFile(Object a, Object b) {
        return false;
    }

    public void uploadLogFile(Object a) {}

    public String getStackFoldPrefix() {
        return "";
    }

    public long getConfigDumpIntervalMillis() {
        return 0;
    }
}

File: noop/src/main/kotlin/me/chunyu/yuriel/kotdebugtool/ui/BlockCanaryContext.kt
package me.chunyu.yuriel.kotdebugtool.ui

import android.content.Context
import java.io.File

/**
 * Created by yuriel on 8/30/16.
 */
open class BlockCanaryContext(any: Any) {

    open val configDuration: Int = 99999

    open val configBlockThreshold: Long = 0L

    open val isNeedDisplay: Boolean = false

    open val qualifier: String = ""

    open val uid: String = ""

    open val networkType: String = ""

    open val context: Context? = null

    open val logPath: String = ""

    open fun zipLogFile(src: Array<File>?, dest: File): Boolean = false

    open fun uploadLogFile(zippedFile: File) {}

    open val stackFoldPrefix: String? = null

    open val configDumpIntervalMillis: Long = 0L
}